### ISSUE TITLE
Build breaking under npm version <7.x. uslug package URL is locked to private repository

### ISSUE THREAD

**Issue by ameuret** [laurent22/joplin#5376]
When I run `npm install` from the top repository folder, it fails with the error:

```
npm ERR! Error while executing:
npm ERR! /usr/bin/git ls-remote -h -t ssh://git@github.com/laurent22/uslug.git
npm ERR!
npm ERR! git@github.com: Permission denied (publickey).
npm ERR! fatal: Could not read from remote repository.
```

## Workaround :smiley: 

As soon as I fix the URL in `./packages/lib/package-lock.json` and `./packages/renderer/package-lock.json`, from the current value to: `https://github.com/laurent22/uslug.git`, the build passes :+1: .

## Environment

Joplin version: 2.3.5
Platform: Linux
OS specifics: Manjaro
npm: 6.14.13
lerna: 4.0.0

**Comment by laurent22** [comment]
The package value is fine - https://github.com/laurent22/joplin/blob/814f602bd66a7ec2b1b4c6ca63b24f97066031fa/packages/renderer/package.json#L49 

Where did you get the code from?

**Comment by roman-r-m** [comment]
I'm seeing something similar in `lib/package-lock.json` and `renderer/package-lock.json` after running `npm i`:
```
[joplin]$ rg uslug | grep ssh
packages/lib/package-lock.json:                 "resolved": "git+ssh://git@github.com/laurent22/uslug.git#ba2834d79beb0435318709958b2f5e817d96674d",
packages/lib/package-lock.json:                 "version": "git+ssh://git@github.com/laurent22/uslug.git#ba2834d79beb0435318709958b2f5e817d96674d",
packages/renderer/package-lock.json:                    "resolved": "git+ssh://git@github.com/laurent22/uslug.git#ba2834d79beb0435318709958b2f5e817d96674d",
packages/renderer/package-lock.json:                    "version": "git+ssh://git@github.com/laurent22/uslug.git#ba2834d79beb0435318709958b2f5e817d96674d",
```

**Comment by laurent22** [comment]
Thanks, that's strange a particular url format would be used in package.json and a different one in package-lock. Maybe a bug in the v2 format.

**Comment by roman-r-m** [comment]
This looks related https://github.com/npm/cli/issues/2054
It's closed but not sure in which version

**Comment by roman-r-m** [comment]
Strangely, I don't have this issue. Not sure why.

**Comment by roman-r-m** [comment]
Ah, maybe this is why

>  npm: 6.14.13

@ameuret - try npm 7

**Comment by ameuret** [comment]
Good catch @roman-r-m ! Indeed running under a npm 7 environment does not trigger the issue.

**Comment by laurent22** [comment]
Thanks for checking. I wonder if it happens every time with npm 6 or if it's just with certain configurations. If it's always, we should probably enforce the npm 7 requirement in package.json.

**Comment by ameuret** [comment]
I got to it through the arch joplin package during a routine update. IIRC I also had to install lerna. Then I stumbled upon this discrepancy between the `package` and its lockfile. I guess that specifying lerna as a prereq makes sense, at least in the doc so that package maintainers enforce it. As for the dependency on npm 7, now that I think of it it may be unnecessary after all, I fast-forwarded from 6.14 to 7.21 so I can't tell much. I'm no npm expert but I'd say the root cause remains. It's not helpful that npm now conceals it.
